.github/workflows/main.yaml

```
@@ -17,31 +18,81 @@ jobs:
+          # Check if coverage meets minimum 70%
+          COVERAGE_LINE=$(lcov --summary coverage/integration_lcov.info | grep "lines" | sed 's/.*lines.......: \([0-9.]*%\).*/\1/')
+          COVERAGE_NUMBER=$(echo ${COVERAGE_LINE} | sed 's/%//')
+          if (( $(echo "${COVERAGE_NUMBER} >= 70" | bc -l) )); then
+            echo "Integration test coverage is acceptable: ${COVERAGE_LINE} (minimum: 70%)"
+          else
+            echo "Integration test coverage is below acceptable levels: ${COVERAGE_LINE} (minimum: 70%)"
+            exit 1
+          fi
+
+      - name: Upload integration coverage report
+        uses: actions/upload-artifact@v2
```

actionlint (1.7.4)

the runner of "actions/upload-artifact@v2" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)